src/ModelContextProtocol/Client/McpClient.cs

```diff
@@ -42,7 +42,7 @@ public McpClient(IClientTransport clientTransport, McpClientOptions options, Mcp
 
             SetRequestHandler<CreateMessageRequestParams, CreateMessageResult>(
                 RequestMethods.SamplingCreateMessage,
-                (request, ct) => samplingHandler(request, ct));
+                (request, ct) => samplingHandler(request, new ClientTokenProgress(this, request?.Meta?.ProgressToken), ct));
```

Comment:
This should probably follow the same pattern we do on the server side, where if `request?.Meta?.ProgressToken` is null, we instead use some NullTokenProgress.Instance singleton.